files/uncompress.py

```diff
@@ -0,0 +1,265 @@
+options:
+  src:
+    description:
+      - If copy=yes (default), local path to compressed file to copy to the target server; can be absolute or relative. If copy=no, path on the target server to existing compressed file to unpack.
+      - If copy=no and src contains ://, the remote machine will download the file from the url first.
+    required: true
+    default: null
+  dest:
+    description:
+      - Remote absolute path where the file should be uncompressed.
+    required: true
+    default: null
+  copy:
+    description:
+      - "If true, the file is copied from local 'master' to the target machine, otherwise, the plugin will look for src file at the target machine."
+    required: false
+    choices: [ "yes", "no" ]
+    default: "yes"
+  deep_check:
+    description:
+      - "If true, and dest already exists, the file performs a longer and more extensive test than just filesize before deciding to overwrite or not"
+    required: false
+    choices: [ "yes", "no" ]
+    default: "no"
+author: "Jonathan Mainguy (@Jmainguy)"
+notes:
+    - requires C(file)/C(xz) commands on target host
+    - requires gzip and bzip python modules
+    - can handle I(gzip), I(bzip2) and I(xz) compressed files
+    - detects type of compressed file automatically
+'''
+
+EXAMPLES = '''
+# Uncompress foo.gz to /tmp/foo
+- uncompress: src=foo.gz dest=/tmp/foo
+
+# Uncompress a file that is already on the remote machine
+- uncompress: src=/tmp/foo.xz dest=/usr/local/bin/foo copy=no
+
+# Uncompress a file that needs to be downloaded
+- uncompress: src=https://example.com/example.bz2 dest=/usr/local/bin/example copy=no
+'''
+
+import os
+import shutil
+import gzip
+import bz2
+import filecmp
+
+# When downloading an archive, how much of the archive to download before
+# saving to a tempfile (64k)
+BUFSIZE = 65536
+
+def ungzip(src, dest):
+    """
+    Uncompress gzip files.
+    """
+    try:
+        f_out = open(dest, 'wb')
+        f_in = gzip.open(src, 'rb')
+        try:
+            shutil.copyfileobj(f_in, f_out)
+        finally:
+            f_out.close()
+            f_in.close()
+        msg=""
+    except Exception, e:
+        msg="%s" % e
+
+    return msg
+
+def unbzip(src, dest):
+    """
+    Uncompress bzip files.
+    """
+    try:
+        f_out = open(dest, 'wb')
+        f_in = bz2.BZ2File(src, 'rb')
+        try:
+            shutil.copyfileobj(f_in, f_out)
+        finally:
+            f_out.close()
+            f_in.close()
+        msg=""
+    except Exception, e:
+        msg="%s" % e
+
+    return msg
+
+def unxzip(module, src, dest):
+    """
+    Uncompress xz files. Since we must support python 2.4 (EL 5) we cannot import lzma and use native python.
+    Also, xz does not support a destination flag, so we must use > to redirect stdout to file. Really lame.
+    """
+    cmd_path = module.get_bin_path('xz')
+    cmd = '%s -c -d %s > %s' % (cmd_path, src, dest)
```

Comment:
Shouldn't we escape the path for shell metacharacters ? This seems rather unsafe, and while the stuff from a playbook is likely under a sysadmin control, we can't be sure someone will not take data coming from a non admin controlled location.